GliaWidgets/SecureConversations/SecureConversations.ChatWithTranscriptModel.swift

```diff
@@ -180,43 +156,214 @@ extension SecureConversations {
+
         var action: ActionCallback?
         var delegate: DelegateCallback?
 
         var engagementAction: EngagementViewModel.ActionCallback?
         var engagementDelegate: EngagementViewModel.DelegateCallback?
 
+        private let downloader: FileDownloader
+
+        private let isCustomCardSupported: Bool
+
+        private var isViewLoaded: Bool = false
+
+        private let environment: Environment
+
+        private let sections = [
+            Section<ChatItem>(0)
+        ]
+
+        var historySection: Section<ChatItem> { sections[0] }
+
         var numberOfSections: Int {
-            // TODO: MOB-1863
-            .zero
+            sections.count
         }
 
+        init(
+            isCustomCardSupported: Bool,
+            environment: Environment
+        ) {
+            self.isCustomCardSupported = isCustomCardSupported
+            self.environment = environment
+            self.downloader = FileDownloader(
+                environment: .init(
+                    fetchFile: environment.fetchFile,
+                    fileManager: environment.fileManager,
+                    data: environment.data,
+                    date: environment.date,
+                    gcd: environment.gcd,
+                    localFileThumbnailQueue: environment.localFileThumbnailQueue,
+                    uiImage: environment.uiImage,
+                    createFileDownload: environment.createFileDownload
+                )
+            )
+        }
+
+        // TODO: Common with chat model, should it be unified?
         func numberOfItems(in section: Int) -> Int {
-            // TODO: MOB-1863
-            .zero
+            sections[section].itemCount
         }
 
         func item(for row: Int, in section: Int) -> ChatItem {
-            // TODO: MOB-1863
-            fatalError("Unimplemnted")
+            let section = sections[section]
+            let item = section[row]
+
+            switch item.kind {
+            case .operatorMessage(let message, _, _):
+                message.downloads = downloader.downloads(
+                    for: message.attachment?.files,
+                    autoDownload: .images
+                )
+                if shouldShowOperatorImage(for: row, in: section) {
+                    let imageUrl = message.operator?.pictureUrl
+                    let kind: ChatItem.Kind = .operatorMessage(
+                        message,
+                        showsImage: true,
+                        imageUrl: imageUrl
+                    )
+                    return ChatItem(kind: kind)
+                }
+                return item
+            case .visitorMessage(let message, _):
+                message.downloads = downloader.downloads(
+                    for: message.attachment?.files,
+                    autoDownload: .images
+                )
+                return item
+            case .choiceCard(let message, _, _, let isActive):
```

Comment:
So this model is going to be a unified model or its own? Because if it will be its own, then it doesn't need choice/custom card handling.

Reply:
But choice cards are a part of history, right? In that case they still need to be rendered.

Reply:
Yep, makes sense.